services/chrome-web-store/chrome-web-store-version.tester.js

```diff
@@ -12,7 +13,16 @@ t.create('Version (not found)')
   .expectBadge({ label: 'chrome web store', message: 'not found' })
 
 // Keep this "inaccessible" test, since this service does not use BaseService#_request.
+const mockAgent = new MockAgent()
 t.create('Version (inaccessible)')
   .get('/alhjnofcnnpeaphgeakdhkebafjcpeae.json')
-  .networkOff()
+  // webextension-store-meta uses undici internally, so we can't mock it with nock
+  .before(function () {
+    setGlobalDispatcher(mockAgent)
+    mockAgent.disableNetConnect()
+  })
+  .after(async function () {
+    await mockAgent.close()
+    setGlobalDispatcher(new Agent())
+  })
```

Comment:
Just for code archaeology purposes:

Uniquely, the chrome web store integration bypasses our usual HTTP client. We farm out making the request to an external library.

In v1.2.0, webextension-store-meta switched their HTTP client from node-fetch to undici.

In theory, that is just internals not API surface but because undici bypasses the normal node HTTP stack, this means we now can't mock the requests it makes using nock (see https://github.com/nock/nock/issues/2183 ). Oh no - they removed [spacebar heating](https://xkcd.com/1172/)!

For the purposes of just getting these tests to work, I've implemented an ad-hoc mocking solution for undici and copied and pasted it to the 3 places we need to do this. If we had a lot of code using undici, I'd pull this out to an abstraction but given this is a uniquely non-standard service we can live with this.

Reply:
Yep, it intentionally bypasses the node `http` module because it has problems handling non-ascii `location` headers sent by Google's server.